Issue #3455464 - Transform GroupMembershipRequestForm to extend the ContentEntityForm, so we can use field API instead of a custom form

nkoporec opened the pull request:
## Problem
In issue https://www.drupal.org/project/social/issues/3453509 we discussed removing the hook to automatically add the field_grequest_message. In order for that change to work we also need to remove any "hard-coded" assumptions that this field exist, since the hook will be remove we can no longer assume that the field will always be there. 

This is a breaking change. 

## Solution
In  https://github.com/goalgorilla/open_social/pull/3923 the idea from @Kingdutch  was to refactor and use ContentEntityForm as base. We need to change from FormBase to ContentEntityForm and also update the controller to load the entity form. I also created a new entity form definition since our logic requires a bit of a different flow then the join-group defined by the group_content entity, instead of writing a hook_form_alter for this, I opted to just define a new entity form type and use that in the controller.

## Issue tracker
https://www.drupal.org/project/social/issues/3455464

## Theme issue tracker
<!-- *[Required if applicable] Paste a link to the drupal.org theme issue queue item, either from [socialbase](https://www.drupal.org/project/socialbase) or [socialblue](https://www.drupal.org/project/socialblue). If any other issue trackers were used, include links to those too.* -->

## How to test
1. Create a new group type and set the option to request to join for all users.
2. As a seperate user request the access to join the group and write a request message
3. As the group admin check if any request has been made and verify that the message is present.

## Screenshots
<!-- *[Required if new feature, and if applicable] If this Pull Request makes visual changes then please include some screenshots that show what has changed here. A before and after screenshot helps the reviewer determine what changes were made.* -->

## Release notes
Refactored the GroupMembershipRequestForm to use the ContentEntityForm, this gives us to option to use the field API from core.

## Change Record
<!-- *[Required if applicable] If this Pull Request changes the way that developers should do things or introduces a new API for developers then a change record to document this is needed. Please provide a draft for a change record or a link to an unpublished change record below. Existing change records can be consulted as example. Please provide a draft for a change record or a link to an unpublished change record below. [Existing change records](https://www.drupal.org/list-changes/social) can be consulted as example.* -->

## Translations
<!--
*[Optional]Translatable strings are always extracted from the latest development branch. To ensure translations remain available for platforms running older versions of Open Social the original string should be added to `translations.php` when it's changed or removed.*
- [ ] Changed or removed source strings are added to the `translations.php` file.
-->

open-social-tugboat commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3937-qvi9f50sgk0fwumx6c7meppc1dz6nno6.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6671a2366939b442cabfea31

open-social-tugboat commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3937-qvi9f50sgk0fwumx6c7meppc1dz6nno6.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6671a2366939b442cabfea31

open-social-tugboat commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3937-qvi9f50sgk0fwumx6c7meppc1dz6nno6.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6671a2366939b442cabfea31

open-social-tugboat commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3937-qvi9f50sgk0fwumx6c7meppc1dz6nno6.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6671a2366939b442cabfea31

open-social-tugboat commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3937-qvi9f50sgk0fwumx6c7meppc1dz6nno6.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6671a2366939b442cabfea31

open-social-tugboat commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3937-qvi9f50sgk0fwumx6c7meppc1dz6nno6.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6671a2366939b442cabfea31

open-social-tugboat commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3937-qvi9f50sgk0fwumx6c7meppc1dz6nno6.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6671a2366939b442cabfea31

Kingdutch reviewed (changes requested):
@nkoporec Hey Nejc, you say the `GroupRelationship` is a config entity but as far as I can find that's not true and it's a content entity.

`Drupal\group\Entity\GroupRelationship` is the content entity that's instantiated by `grequest` as part of the group membership request (which is a GroupRelationType in `Drupal\grequest\Plugin\Group\Relation\GroupMembershipRequest` and the `grequest` module defines the fields on the `group_content` content entity.

The `GroupRelationship` is also the entity instantiated by `$this->entityTypeManager->getStorage('group_content')->create`. So the form is already creating a content entity. 

Can you elaborate on what you were running into and where the config entity comes from?

open-social-tugboat commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3937-qvi9f50sgk0fwumx6c7meppc1dz6nno6.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6671a2366939b442cabfea31

open-social-tugboat commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3937-qvi9f50sgk0fwumx6c7meppc1dz6nno6.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6671a2366939b442cabfea31

open-social-tugboat commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3937-qvi9f50sgk0fwumx6c7meppc1dz6nno6.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6671a2366939b442cabfea31

open-social-tugboat commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3937-qvi9f50sgk0fwumx6c7meppc1dz6nno6.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6671a2366939b442cabfea31

open-social-tugboat commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3937-qvi9f50sgk0fwumx6c7meppc1dz6nno6.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6671a2366939b442cabfea31

open-social-tugboat commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3937-qvi9f50sgk0fwumx6c7meppc1dz6nno6.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6671a2366939b442cabfea31

open-social-tugboat commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3937-qvi9f50sgk0fwumx6c7meppc1dz6nno6.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6671a2366939b442cabfea31

open-social-tugboat commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3937-qvi9f50sgk0fwumx6c7meppc1dz6nno6.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6671a2366939b442cabfea31

open-social-tugboat commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3937-qvi9f50sgk0fwumx6c7meppc1dz6nno6.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6671a2366939b442cabfea31

open-social-tugboat commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3937-qvi9f50sgk0fwumx6c7meppc1dz6nno6.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6671a2366939b442cabfea31

open-social-tugboat commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3937-qvi9f50sgk0fwumx6c7meppc1dz6nno6.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6671a2366939b442cabfea31

open-social-tugboat commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3937-qvi9f50sgk0fwumx6c7meppc1dz6nno6.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6671a2366939b442cabfea31

tbsiqueira commented:
Hi @nkoporec , please update the PR description, solution, to match changes that you did at the code.

tbsiqueira commented on modules/social_features/social_group/modules/social_group_request/src/Form/GroupRequestMembershipRequestForm.php:
```diff
@@ -99,56 +55,66 @@ public function getFormId() {
+    /** @var \Drupal\group\Entity\GroupRelationshipInterface $entity */
+    $entity = $this->entity;
 
-    $this->cacheTagsInvalidator->invalidateTags(['request-membership:' . $this->group->id()]);
```
We are not taking care of cache invalidation anywhere else, this will create a bug, can you check this?

Kingdutch replied:
Good catch. I would've actually expected this not to break anything, but it seems in Open Social we've created our own cache-tag where we shouldn't have. It's used in a couple of places so we'll need to keep it around for now. However, we should post a follow-up to move to the tags provided by the group module instead then it happens automatically for us.

From `GroupRelationship::getListCacheTagsToInvalidate` there's already very precise tags for different types of lists <3 We likely want the following one
```
    // A specific group gets any content using a specific plugin.
    // E.g.: A group's list of members can be flushed with this.
    $tags[] = "group_content_list:plugin:$plugin_id:group:$group_id";
```

tbsiqueira commented on modules/social_features/social_group/modules/social_group_request/src/Form/GroupRequestMembershipRequestForm.php:
```diff
@@ -2,90 +2,46 @@
-   *
-   * @var \Drupal\group\Entity\GroupRelationshipInterface
-   */
-  protected $groupContent;
```
This change is irrelevant to this PR as this was not previously used, let's split this change on a separated PR

tbsiqueira commented on modules/social_features/social_group/modules/social_group_request/src/Form/GroupRequestMembershipRequestForm.php:
```diff
@@ -2,90 +2,46 @@
-   *
-   * @var \Drupal\Core\Cache\CacheTagsInvalidatorInterface
-   */
-  protected $cacheTagsInvalidator;
```
We are removing the cacheTagInvalidator but I don't see a replacement

tbsiqueira commented on modules/social_features/social_group/modules/social_group_request/src/Form/GroupRequestMembershipRequestForm.php:
```diff
@@ -2,90 +2,46 @@
-   *   The entity type manager.
+   * {@inheritdoc}
    */
-  public function __construct(
```
Let's make sure not to remove the constructor because this now causes tests to be unable to instantiate this class fully using mocked services.

tbsiqueira commented:
This has the potential to be a BC breaking change, can you check if any of our extensions extended this class? I know that in Cablecar we do with at least two classes.

@nkoporec

Kingdutch commented on modules/social_features/social_group/modules/social_group_request/src/Form/GroupRequestMembershipRequestForm.php:
```diff
@@ -99,56 +55,66 @@ public function getFormId() {
-      '#tag' => 'p',
-      '#value' => $this->t('You can leave a message in your request. Only when your request is approved, you will receive a notification via email and notification center.'),
-    ];
+    /** @var \Drupal\group\Entity\Storage\GroupRelationshipTypeStorageInterface $storage */
+    $storage = $this->entityTypeManager->getStorage('group_content_type');
+    $group_type_id = (string) $group->getGroupType()->id();
+    $relation_type_id = $storage->getRelationshipTypeId($group_type_id, 'group_membership_request');
+    $group_content = $this->entityTypeManager->getStorage('group_content')->create([
+      'type' => $relation_type_id,
+      'gid' => $group->id(),
+      'entity_id' => $this->currentUser()->id(),
+      'grequest_status' => GroupMembershipRequest::REQUEST_PENDING,
+    ]);
+    $this->setEntity($group_content);
 
-    $form['message'] = [
-      '#type' => 'textarea',
-      '#title' => $this->t('Message'),
-    ];
+    $form = parent::buildForm($form, $form_state);
```
Can you explain why we still need this? I would expect all of this to now be taken care of by Drupal's Content Entity Form?

Kingdutch commented on modules/social_features/social_group/modules/social_group_request/src/Form/GroupRequestMembershipRequestForm.php:
```diff
@@ -99,56 +55,66 @@ public function getFormId() {
-      '#value' => $this->t('Send request'),
-      '#button_type' => 'primary',
-    ];
+    if (isset($form['field_grequest_message'])) {
+      $form['description'] = [
+        '#type' => 'html_tag',
+        '#tag' => 'p',
+        '#value' => $this->t('You can leave a message in your request. Only when your request is approved, you will receive a notification via email and notification center.'),
+        '#weight' => -10,
+      ];
+    }
```
I think this is something we can instead move into `field.field.group_content.*.field_grequest_message` under the `description` string, right? Or does that not yield the same output?

Is there another way we could add that? The answer can also be yes but we do it in a follow-up. But I'd love to see if we can make this very specific class be mostly gone.

Kingdutch commented on modules/social_features/social_group/modules/social_group_request/src/Form/GroupRequestMembershipRequestForm.php:
```diff
@@ -99,56 +55,66 @@ public function getFormId() {
-    $storage = $this->entityTypeManager->getStorage('group_content_type');
-    $group_type_id = (string) $this->group->getGroupType()->id();
-    $relation_type_id = $storage->getRelationshipTypeId($group_type_id, 'group_membership_request');
+  protected function actions(array $form, FormStateInterface $form_state) {
+    $actions = parent::actions($form, $form_state);
+    $actions['submit']['#value'] = $this->t('Send request');
+    return $actions;
+  }
```
Nice :D

Kingdutch commented on modules/social_features/social_group/modules/social_group_request/src/Form/GroupRequestMembershipRequestForm.php:
```diff
@@ -99,56 +55,66 @@ public function getFormId() {
-    else {
-      $this->messenger()->addError($this->t('Error when creating a request to join'));
-    }
+    /** @var \Drupal\group\Entity\GroupRelationshipInterface $entity */
+    $entity = $this->entity;
```
Prefer using assert here. See [development guideline 2.2 assert over `@var`](https://open-social.slite.com/app/docs/dm052z21PPpHST#02da03b7) for details.

Kingdutch commented on modules/social_features/social_group/modules/social_group_request/src/Form/GroupRequestMembershipRequestForm.php:
```diff
@@ -2,90 +2,46 @@
-    $this->currentUser = $current_user;
-    $this->setStringTranslation($string_translation);
-    $this->entityTypeManager = $entity_type_manager;
+  public static function create(ContainerInterface $container) {
+    $form = parent::create($container);
+    $form->moduleHandler = $container->get('module_handler');
+    $form->entityTypeManager = $container->get('entity_type.manager');
+    return $form;
```
A static `create` function shouldn't do anything beyond calling `__construct`. Not doing so can lead to services being needed which can not be injected through a constructor which blocks testing with PHPUnit.

Additionally in this case `moduleHandler` and `entityTypeManager` are both already provided by `EntityForm` so we can likely remove them from the class altogether.

Kingdutch commented on modules/social_features/social_group/modules/social_group_request/social_group_request.install:
```diff
@@ -105,3 +107,18 @@ function social_group_request_update_13002(): void {
+ * Apply the new entity form display for group content type.
+ */
+function social_group_request_update_13003(): void {
+  $config_path = \Drupal::service('extension.list.module')->getPath('social_group_request') . '/config/static';
+  $source = new FileStorage($config_path);
+  /** @var \Drupal\Core\Config\StorageInterface $config_storage */
+  $config_storage = \Drupal::service('config.storage');
+
+  $data = $source->read('core.entity_form_display.group_content.group_content_type_7fcb76fdf61a9.default_13003');
+  if (is_array($data)) {
+    $config_storage->write('core.entity_form_display.group_content.group_content_type_7fcb76fdf61a9.default', $data);
+  }
```
Can we switch this over to be consistent and use the update helper module instead?

Kingdutch reviewed (changes requested):
I don't quite understand the logic that's now in `buildForm` and I'd love to understand it before we merge this :) There's also a few other things where there's room for improvement, or at least clarification for potential follow-ups or people touching this code in the future.

open-social-tugboat commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3937-qvi9f50sgk0fwumx6c7meppc1dz6nno6.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6671a2366939b442cabfea31

open-social-tugboat commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3937-qvi9f50sgk0fwumx6c7meppc1dz6nno6.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6671a2366939b442cabfea31

open-social-tugboat commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3937-qvi9f50sgk0fwumx6c7meppc1dz6nno6.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6671a2366939b442cabfea31

open-social-tugboat commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3937-qvi9f50sgk0fwumx6c7meppc1dz6nno6.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6671a2366939b442cabfea31

open-social-tugboat commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3937-qvi9f50sgk0fwumx6c7meppc1dz6nno6.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6671a2366939b442cabfea31

open-social-tugboat commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3937-qvi9f50sgk0fwumx6c7meppc1dz6nno6.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6671a2366939b442cabfea31

open-social-tugboat commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3937-qvi9f50sgk0fwumx6c7meppc1dz6nno6.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6671a2366939b442cabfea31

open-social-tugboat commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3937-qvi9f50sgk0fwumx6c7meppc1dz6nno6.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6671a2366939b442cabfea31

nkoporec commented:
@tbsiqueira @Kingdutch  most of your remarks has been fixed and the logic has been updated, so please perform a full re-review again please.

Some explanation of my changes:

1. I created a new form definition for the group_content entity, reasons for this was that the group_content entity defines the  'group-join' form which loads the `core.entity_form_display.group_content.group_content_type_7fcb76fdf61a9.default` config,  but this does not give us all the functionality that we had before with the custom form, mainly the cache clearing and the redirects after submit, we could use a hook_form_alter but this seemed like a better fit for this use-case.

2. In the controller GroupRequestController.php we are loading the form via Ajax with the formBuilder this was now changed and we are now using the entity form builder, this resolved the issue that I had with the entity not being set correctly in the form.

3. I moved to update_helper module for the updates, there is one config that still requires the static method, the core.entity_form_display.group_content.group_content_type_7fcb76fdf61a9.default config does not exist (since it was previously done with a hook), so we need to create the config and afaik this is not possible with the update_helper module, since it only does the updates.

open-social-tugboat commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3937-qvi9f50sgk0fwumx6c7meppc1dz6nno6.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6671a2366939b442cabfea31

open-social-tugboat commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3937-qvi9f50sgk0fwumx6c7meppc1dz6nno6.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6671a2366939b442cabfea31

open-social-tugboat commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3937-qvi9f50sgk0fwumx6c7meppc1dz6nno6.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6671a2366939b442cabfea31

open-social-tugboat commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3937-qvi9f50sgk0fwumx6c7meppc1dz6nno6.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6671a2366939b442cabfea31

open-social-tugboat commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3937-qvi9f50sgk0fwumx6c7meppc1dz6nno6.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6671a2366939b442cabfea31

open-social-tugboat commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3937-qvi9f50sgk0fwumx6c7meppc1dz6nno6.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6671a2366939b442cabfea31

open-social-tugboat commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3937-qvi9f50sgk0fwumx6c7meppc1dz6nno6.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6671a2366939b442cabfea31

open-social-tugboat commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3937-qvi9f50sgk0fwumx6c7meppc1dz6nno6.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6671a2366939b442cabfea31

open-social-tugboat commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3937-qvi9f50sgk0fwumx6c7meppc1dz6nno6.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6671a2366939b442cabfea31

open-social-tugboat commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3937-qvi9f50sgk0fwumx6c7meppc1dz6nno6.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6671a2366939b442cabfea31

open-social-tugboat commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3937-qvi9f50sgk0fwumx6c7meppc1dz6nno6.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6671a2366939b442cabfea31

open-social-tugboat commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3937-qvi9f50sgk0fwumx6c7meppc1dz6nno6.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6671a2366939b442cabfea31

open-social-tugboat commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3937-qvi9f50sgk0fwumx6c7meppc1dz6nno6.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6671a2366939b442cabfea31

open-social-tugboat commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3937-qvi9f50sgk0fwumx6c7meppc1dz6nno6.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6671a2366939b442cabfea31

open-social-tugboat commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3937-qvi9f50sgk0fwumx6c7meppc1dz6nno6.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6671a2366939b442cabfea31

open-social-tugboat commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3937-qvi9f50sgk0fwumx6c7meppc1dz6nno6.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6671a2366939b442cabfea31

open-social-tugboat commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3937-qvi9f50sgk0fwumx6c7meppc1dz6nno6.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6671a2366939b442cabfea31

Kingdutch commented:
This is instead solved by #3968
